master/internal/api_experiment.go

```diff
@@ -1565,12 +1601,19 @@ func (a *apiServer) ContinueExperiment(
 		// TODO consider moving this to trial_id_task_id or some other level to preserve
 		// the history of what happened during the trial. We should also do this
 		// with submitted config yamls likely and display these in the webui.
-		if _, err := tx.NewUpdate().Model(&model.Trial{}).
-			Set("restarts = 0").
-			Set("end_time = null").
-			Where("id = ?", trialsResp.Trials[0].Id).
-			Exec(ctx); err != nil {
-			return fmt.Errorf("zeroing out trial restarts: %w", err)
+		if len(trialsResp.Trials) > 0 {
+			values := tx.NewValues(&trialsResp.Trials)
+			if _, err := tx.NewUpdate().
+				With("_data", values).
+				Model((*model.Trial)(nil)).
+				TableExpr("_data").
+				Set("restarts = 0").
+				Set("end_time = null").
+				Where("trial.id = _data.id").
+				Exec(ctx); err != nil {
+				return fmt.Errorf("zeroing out trial restarts: %w", err)
+			}
+			e.continueTrials = ptrs.Ptr(true)
```

Comment:
I don't think this actually makes a difference as far as I can tell, but I think I would prefer this to be always set and not just when we have persisted trials

Reply:
done